CMake, Build: -DENABLE_INTERNAL_DAV1D=ON causes configuration error

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug

<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Here is a clear and concise description of what the problem is:
When configuring kodi with the argument -DENABLE_INTERNAL_DAV1D=ON, errors occur.

I suspect that the target "dav1d" is defined twice. Once by "cmake/scripts/common/Macros.cmake:376 (find_package)" and once by "cmake/modules/FindFFMPEG.cmake:34 (find_package)". I have output two messages in the "FindDav1d.cmake", one contains a trace message-type.

The log can be seen in the following.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Kodi should build with internal libdav1d


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
it is not possible to build with -DENABLE_INTERNAL_DAV1D=ON.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->



### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1.  I used following command:
 ```
$cmake ../xbmc/ -DCMAKE_INSTALL_PREFIX=/usr/local -DCORE_PLATFORM_NAME=gbm -DAPP_RENDER_SYSTEM=gles -DENABLE_INTERNAL_FFMPEG=ON -DENABLE_INTERNAL_DAV1D=ON
``` 
3.
4.


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:
[cmake-dav1d-error.txt](https://github.com/xbmc/xbmc/files/12567955/cmake-dav1d-error.txt)


### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->



## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: current alpha3 and master



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

Indeed.  Having removed -DENABLE_INTERNAL_DAV1D, I now have an error with FFMPEG.
```-- Could NOT find Dav1d (missing: DAV1D_LIBRARY DAV1D_INCLUDE_DIR) (Required is at least version "1.1.0")
-- dav1d not found, internal ffmpeg build will be missing AV1 support!
CMake Error at cmake/modules/FindFFMPEG.cmake:127 (add_library):
  add_library cannot create imported target "ffmpeg::libavcodec" because
  another target with the same name already exists.
Call Stack (most recent call first):
  cmake/modules/FindFFMPEG.cmake:387 (buildFFMPEG)
  cmake/scripts/common/Macros.cmake:376 (find_package)
  cmake/scripts/common/Macros.cmake:390 (find_package_with_ver)
  CMakeLists.txt:226 (core_require_dep)```

@fuzzard or anyone else that might have a quick fix!

--- a/cmake/modules/FindDav1d.cmake
+++ b/cmake/modules/FindDav1d.cmake
@@ -3,60 +3,67 @@
 # --------
 # Finds the dav1d library
 #
-# This will define the following variables::
+# This will define the following target:
 #
-# DAV1D_FOUND - system has dav1d
-# DAV1D_INCLUDE_DIRS - the dav1d include directories
-# DAV1D_LIBRARIES - the dav1d libraries
+#   dav1d::dav1d   - The dav1d library
 
-if(ENABLE_INTERNAL_DAV1D)
-  include(cmake/scripts/common/ModuleHelpers.cmake)
+if(NOT TARGET dav1d::dav1d)
+  if(ENABLE_INTERNAL_DAV1D)
+    include(cmake/scripts/common/ModuleHelpers.cmake)
 
-  set(MODULE_LC dav1d)
+    set(MODULE_LC dav1d)
 
-  SETUP_BUILD_VARS()
+    SETUP_BUILD_VARS()
 
-  set(DAV1D_VERSION ${${MODULE}_VER})
+    set(DAV1D_VERSION ${${MODULE}_VER})
 
-  find_program(NINJA_EXECUTABLE ninja REQUIRED)
-  find_program(MESON_EXECUTABLE meson REQUIRED)
+    find_program(NINJA_EXECUTABLE ninja REQUIRED)
+    find_program(MESON_EXECUTABLE meson REQUIRED)
 
-  set(CONFIGURE_COMMAND ${MESON_EXECUTABLE}
-                        --buildtype=release
-                        --default-library=static
-                        --prefix=${DEPENDS_PATH}
-                        --libdir=lib
-                        -Denable_asm=true
-                        -Denable_tools=false
-                        -Denable_examples=false
-                        -Denable_tests=false
-                        ../dav1d)
-  set(BUILD_COMMAND ${NINJA_EXECUTABLE})
-  set(INSTALL_COMMAND ${NINJA_EXECUTABLE} install)
+    set(CONFIGURE_COMMAND ${MESON_EXECUTABLE}
+                          --buildtype=release
+                          --default-library=static
+                          --prefix=${DEPENDS_PATH}
+                          --libdir=lib
+                          -Denable_asm=true
+                          -Denable_tools=false
+                          -Denable_examples=false
+                          -Denable_tests=false
+                          ../dav1d)
+    set(BUILD_COMMAND ${NINJA_EXECUTABLE})
+    set(INSTALL_COMMAND ${NINJA_EXECUTABLE} install)
 
-  BUILD_DEP_TARGET()
-else()
-  if(PKG_CONFIG_FOUND)
-    pkg_check_modules(PC_DAV1D dav1d QUIET)
+    BUILD_DEP_TARGET()
+  else()
+    if(PKG_CONFIG_FOUND)
+      pkg_check_modules(PC_DAV1D dav1d QUIET)
+    endif()
+
+    find_library(DAV1D_LIBRARY NAMES dav1d libdav1d
+                               PATHS ${PC_DAV1D_LIBDIR}
+                               NO_CACHE)
+
+    find_path(DAV1D_INCLUDE_DIR NAMES dav1d/dav1d.h
+                                PATHS ${PC_DAV1D_INCLUDEDIR}
+                                NO_CACHE)
+
+    set(DAV1D_VERSION ${PC_DAV1D_VERSION})
   endif()
 
-  find_library(DAV1D_LIBRARY NAMES dav1d libdav1d
-                             PATHS ${PC_DAV1D_LIBDIR})
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(Dav1d
+                                    REQUIRED_VARS DAV1D_LIBRARY DAV1D_INCLUDE_DIR
+                                    VERSION_VAR DAV1D_VERSION)
 
-  find_path(DAV1D_INCLUDE_DIR NAMES dav1d/dav1d.h
-                              PATHS ${PC_DAV1D_INCLUDEDIR})
+  if(DAV1D_FOUND)
+    add_library(dav1d::dav1d UNKNOWN IMPORTED)
+    set_target_properties(dav1d::dav1d PROPERTIES
+                                             IMPORTED_LOCATION "${DAV1D_LIBRARY}"
+                                             INTERFACE_INCLUDE_DIRECTORIES "${DAV1D_INCLUDE_DIR}")
+    set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP dav1d::dav1d)
 
-  set(DAV1D_VERSION ${PC_DAV1D_VERSION})
+    if(TARGET dav1d)
+      add_dependencies(dav1d::dav1d dav1d)
+    endif()
+  endif()
 endif()
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Dav1d
-                                  REQUIRED_VARS DAV1D_LIBRARY DAV1D_INCLUDE_DIR
-                                  VERSION_VAR DAV1D_VERSION)
-
-if(DAV1D_FOUND)
-  set(DAV1D_INCLUDE_DIRS ${DAV1D_INCLUDE_DIR})
-  set(DAV1D_LIBRARIES ${DAV1D_LIBRARY})
-endif()
-
-mark_as_advanced(DAV1D_INCLUDE_DIR DAV1D_LIBRARY)

--- a/cmake/modules/FindFFMPEG.cmake
+++ b/cmake/modules/FindFFMPEG.cmake
@@ -32,20 +32,21 @@ macro(buildFFMPEG)
   # Check for dependencies - Must be done before SETUP_BUILD_VARS
   get_libversion_data("dav1d" "target")
   find_package(Dav1d ${LIB_DAV1D_VER} MODULE)
-  if(NOT DAV1D_FOUND)
+  if(NOT TARGET dav1d::dav1d)
     message(STATUS "dav1d not found, internal ffmpeg build will be missing AV1 support!")
+  else()
+    set(FFMPEG_OPTIONS -DENABLE_DAV1D=ON)
   endif()
 
   set(MODULE_LC ffmpeg)
 
   SETUP_BUILD_VARS()
 
-  set(FFMPEG_OPTIONS -DENABLE_CCACHE=${ENABLE_CCACHE}
-                     -DCCACHE_PROGRAM=${CCACHE_PROGRAM}
-                     -DENABLE_VAAPI=${ENABLE_VAAPI}
-                     -DENABLE_VDPAU=${ENABLE_VDPAU}
-                     -DENABLE_DAV1D=${DAV1D_FOUND}
-                     -DEXTRA_FLAGS=${FFMPEG_EXTRA_FLAGS})
+  list(APPEND FFMPEG_OPTIONS -DENABLE_CCACHE=${ENABLE_CCACHE}
+                             -DCCACHE_PROGRAM=${CCACHE_PROGRAM}
+                             -DENABLE_VAAPI=${ENABLE_VAAPI}
+                             -DENABLE_VDPAU=${ENABLE_VDPAU}
+                             -DEXTRA_FLAGS=${FFMPEG_EXTRA_FLAGS})
 
   if(KODI_DEPENDSBUILD)
     set(CROSS_ARGS -DDEPENDS_PATH=${DEPENDS_PATH}
@@ -87,8 +88,8 @@ macro(buildFFMPEG)
 
   BUILD_DEP_TARGET()
 
-  if(TARGET dav1d)
-    add_dependencies(ffmpeg dav1d)
+  if(TARGET dav1d::dav1d)
+    add_dependencies(ffmpeg dav1d::dav1d)
   endif()
 
   find_program(BASH_COMMAND bash)

--- a/cmake/modules/FindFFMPEG.cmake
+++ b/cmake/modules/FindFFMPEG.cmake
@@ -127,37 +127,30 @@ fi")
 
   add_library(ffmpeg::libavcodec INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libavcodec PROPERTIES
-                                           FOLDER "FFMPEG - External Projects"
                                            INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 
   add_library(ffmpeg::libavfilter INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libavfilter PROPERTIES
-                                            FOLDER "FFMPEG - External Projects"
                                             INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 
   add_library(ffmpeg::libavformat INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libavformat PROPERTIES
-                                            FOLDER "FFMPEG - External Projects"
                                             INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 
   add_library(ffmpeg::libavutil INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libavutil PROPERTIES
-                                          FOLDER "FFMPEG - External Projects"
                                           INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 
   add_library(ffmpeg::libswscale INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libswscale PROPERTIES
-                                           FOLDER "FFMPEG - External Projects"
                                            INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 
   add_library(ffmpeg::libswresample INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libswresample PROPERTIES
-                                              FOLDER "FFMPEG - External Projects"
                                               INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 
   add_library(ffmpeg::libpostproc INTERFACE IMPORTED)
   set_target_properties(ffmpeg::libpostproc PROPERTIES
-                                            FOLDER "FFMPEG - External Projects"
                                             INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIR}")
 endmacro()
 
@@ -396,7 +389,6 @@ if(FFMPEG_FOUND)
   if(NOT TARGET ffmpeg::ffmpeg)
     add_library(ffmpeg::ffmpeg INTERFACE IMPORTED)
     set_target_properties(ffmpeg::ffmpeg PROPERTIES
-                                         FOLDER "External Projects"
                                          INTERFACE_INCLUDE_DIRECTORIES "${FFMPEG_INCLUDE_DIRS}"
                                          INTERFACE_COMPILE_DEFINITIONS "${_ffmpeg_definitions}")
   endif()

--- a/cmake/modules/FindFlatBuffers.cmake
+++ b/cmake/modules/FindFlatBuffers.cmake
@@ -44,8 +44,7 @@ if(NOT TARGET flatbuffers::flatbuffers)
 
   add_library(flatbuffers::flatbuffers INTERFACE IMPORTED)
   set_target_properties(flatbuffers::flatbuffers PROPERTIES
-                             FOLDER "External Projects"
-                             INTERFACE_INCLUDE_DIRECTORIES "${FLATBUFFERS_INCLUDE_DIR}")
+                                                 INTERFACE_INCLUDE_DIRECTORIES "${FLATBUFFERS_INCLUDE_DIR}")
 
   add_dependencies(flatbuffers::flatbuffers flatbuffers::flatc)
 

--- a/cmake/modules/FindRapidJSON.cmake
+++ b/cmake/modules/FindRapidJSON.cmake
@@ -64,7 +64,6 @@ if(NOT TARGET RapidJSON::RapidJSON)
   if(RAPIDJSON_FOUND)
     add_library(RapidJSON::RapidJSON INTERFACE IMPORTED)
     set_target_properties(RapidJSON::RapidJSON PROPERTIES
-                                               FOLDER "External Projects"
                                                INTERFACE_INCLUDE_DIRECTORIES "${RAPIDJSON_INCLUDE_DIRS}")
     if(TARGET rapidjson)
       add_dependencies(RapidJSON::RapidJSON rapidjson)

